publish.sh

```diff
@@ -31,13 +23,13 @@ else
     export PYPI_PACKAGE_REPOSITORY="test"
 fi
 
-# export version
-VERSION_LAST_TAG=$(git describe --abbrev=0 --tags 2>/dev/null)
 
-if [ -z "$TRAVIS_TAG" ]; then
-    export PYPI_PACKAGE_VERSION=${VERSION_LAST_TAG}-rc.${TRAVIS_BUILD_NUMBER}
-else 
-    export PYPI_PACKAGE_VERSION=${TRAVIS_TAG}
+if [[ "${EFFECTIVE_BRANCH}" =~ ^(v[0-9]+(\.[0-9]+){0,4})$ ]]; then
+    export PYPI_PACKAGE_VERSION=${EFFECTIVE_BRANCH}
+else
+    git fetch --prune --unshallow --tags
+    VERSION_LAST_TAG=$(git describe --abbrev=0 --tags 2>/dev/null)
+    export PYPI_PACKAGE_VERSION=${VERSION_LAST_TAG}-rc.${RUN_NUMBER}
```

Comment:
For this to be a pre-release version, doesn't the format need to be `{VERSION_LAST_TAG}.dev${RUN_NUMBER}`

Reply:
Even above formate is considered as pre release. I'll keep it as it is just to match it with earlier release.